### PR TITLE
Add labels along with add_marker_layer and write pin to port layer on None

### PR DESCRIPTION
**Pull request by nikosavola**
This PR does three things:
* [Support additional label on](https://github.com/gdsfactory/gdsfactory/commit/1487f4ebb818148d02b00c1ec2308eed9c56db8a) [`add_marker_layer`](https://github.com/gdsfactory/gdsfactory/commit/1487f4ebb818148d02b00c1ec2308eed9c56db8a)
* [Write label to port layer in](https://github.com/gdsfactory/gdsfactory/commit/9dbf88316de4184eab91c4247d8bfe34676751f7) [`add_pins_rectangle_inside` when given `layer_label=None`](https://github.com/gdsfactory/gdsfactory/commit/9dbf88316de4184eab91c4247d8bfe34676751f7)
  * Use the feature in [Strip to ridge taper](https://github.com/gdsfactory/gdsfactory/commit/7e875fc5fd931d70d405bd89023275067fdab4e6) for having CrossSection expanded pins on only final component.

### CODE REVIEW

File: gdsfactory/functions.py [hunk 1]
```diff
@@ -282,7 +283,7 @@ def _inner_decorator(component: ComponentFactory) -> ComponentFactory:
             component: The input function creating a component.
         """
 
-        @cell_with_child
+        @cell_with_child(cache=False)
```
Comment (joamatab) [inline gdsfactory/functions.py:286]: this looks dangerous,

do we need it?

**Review by joamatab — COMMENTED** [review]
i think the way of doing marker layers as a decorator can create mutability issues

why not apply it as a new decorator


```
cell_with_devrec = partial(cell, decorator=add_devrec

```

File: gdsfactory/functions.py (hunk 1 above)
Comment (nikosavola) [reply gdsfactory/functions.py:286]: The tests seemed to fail for me without this so, I guess otherwise the wrapper  is cached

**Comment by nikosavola** [comment]
TODO refactor to in-place version of decorator

File: gdsfactory/functions.py [hunk 2]
```diff
@@ -262,58 +260,43 @@ def add_settings_label(
 
 
 def add_marker_layer(
```
Comment (joamatab) [inline gdsfactory/functions.py:262]: if this is intended to be used a as a decorator it should apply to the component in place

Comment (nikosavola) [reply gdsfactory/functions.py:262]: Good catch, I'll update

**Pull request merged**